tests/integration/elder.test.ts

```diff
@@ -0,0 +1,79 @@
+import Elder, {
+  Model,
+  StringType,
+  type,
+  Adapter,
+  PostgresAdapter,
+  Collection
+} from '../../src'
+
+const config = { adapters: { default: { database: 'ash' } } }
+
+describe('Elder', () => {
+  test('model setup', async () => {
+    class CatModel extends Model {
+      @type('string') name: string
+    }
+    const orm = Elder.create({
+      config,
+      models: { cat: CatModel }
+    })
+    const cats = await CatModel.all()
+    expect(cats).toBeInstanceOf(Collection)
+    expect(cats[0]).toBeInstanceOf(CatModel)
+    expect(cats[0].name).toBe('fluffy')
+    return orm.destroy()
+  })
+
+  test('model with custom type', async () => {
+    class CatModel extends Model {
+      @type('cat-name') name: string
+    }
+    class CatName extends StringType {}
+    const orm = Elder.create({
+      config,
+      models: { cat: CatModel },
+      types: { 'cat-name': CatName }
+    })
+    const cats = await CatModel.all()
+    expect(cats[0].name).toBe('fluffy')
```

Comment:
No, the actual field on the model will be a regular javascript type. I hadn't considered using the (in this case) name getter to return an instance of a type. Is that how rails works? Are there any benefits?

Currently I'm using es proxy to trap any get/setting on the model instance and intervene with my types though this could just as easily be achieved with `Reflect.defineProperty` which I may move to.

Essentially:
```js
myModel.name
// and
myModel.name = 'something'
```
result in (respectively):
```js
get name () {
  const type = Model.meta.attributes['name']
  return type.access(this._name)
}

// and

set name (value) {
  const type = Model.meta.attributes['name']
  this._name = type.modify(value)
  return true
}
```